transports/dns/src/lib.rs

```diff
@@ -110,85 +120,28 @@ const MAX_DNS_LOOKUPS: usize = 32;
 /// result of a single `/dnsaddr` lookup.
 const MAX_TXT_RECORDS: usize = 16;
 
-/// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses
-/// using `async-std` for all async I/O.
-#[cfg(feature = "async-std")]
-pub type DnsConfig<T> = GenDnsConfig<T, AsyncStdResolver>;
-
-/// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses
-/// using `tokio` for all async I/O.
-#[cfg(feature = "tokio")]
-pub type TokioDnsConfig<T> = GenDnsConfig<T, TokioAsyncResolver>;
-
 /// A `Transport` wrapper for performing DNS lookups when dialing `Multiaddr`esses.
+/// You shouldn't need to use this type directly. Use [`tokio::Config`] or [`async_std::Config`] instead.
 #[derive(Debug)]
-pub struct GenDnsConfig<T, R> {
+pub struct Config<T, R> {
     /// The underlying transport.
     inner: Arc<Mutex<T>>,
     /// The DNS resolver used when dialing addresses with DNS components.
     resolver: R,
 }
 
-#[cfg(feature = "async-std")]
-impl<T> DnsConfig<T>
-where
-    T: Send,
-{
-    /// Creates a new [`DnsConfig`] from the OS's DNS configuration and defaults.
-    pub async fn system(inner: T) -> Result<DnsConfig<T>, io::Error> {
-        let (cfg, opts) = system_conf::read_system_conf()?;
-        Self::custom(inner, cfg, opts).await
-    }
-
-    /// Creates a [`DnsConfig`] with a custom resolver configuration and options.
-    pub async fn custom(
-        inner: T,
-        cfg: ResolverConfig,
-        opts: ResolverOpts,
-    ) -> Result<DnsConfig<T>, io::Error> {
-        // TODO: Make infallible in next breaking release. Or deprecation?
-        Ok(DnsConfig {
-            inner: Arc::new(Mutex::new(inner)),
-            resolver: async_std_resolver::resolver(cfg, opts).await,
-        })
-    }
-}
-
-#[cfg(feature = "tokio")]
-impl<T> TokioDnsConfig<T>
-where
-    T: Send,
-{
-    /// Creates a new [`TokioDnsConfig`] from the OS's DNS configuration and defaults.
-    pub fn system(inner: T) -> Result<TokioDnsConfig<T>, io::Error> {
-        let (cfg, opts) = system_conf::read_system_conf()?;
-        Self::custom(inner, cfg, opts)
-    }
-
-    /// Creates a [`TokioDnsConfig`] with a custom resolver configuration
-    /// and options.
-    pub fn custom(
-        inner: T,
-        cfg: ResolverConfig,
-        opts: ResolverOpts,
-    ) -> Result<TokioDnsConfig<T>, io::Error> {
-        // TODO: Make infallible in next breaking release. Or deprecation?
-        Ok(TokioDnsConfig {
-            inner: Arc::new(Mutex::new(inner)),
-            resolver: TokioAsyncResolver::tokio(cfg, opts),
-        })
-    }
-}
+#[deprecated(note = "Use `async_std::Config` or `tokio::Config` instead.")]
+pub type GenDnsConfig<T, R> = Config<T, R>;
 
-impl<T, R> Transport for GenDnsConfig<T, R>
+impl<T, R> Transport for Config<T, R>
```

Comment:
Probably will need to disambiguate this one:

```suggestion
impl<T, R> libp2p_core::Transport for Config<T, R>
```